test: Update the new ZTEST APIs tests/net/traffic_class

hunterhu3000 opened the pull request:
TEST=twister -T tests/net/traffic_class

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

yperess commented on tests/net/traffic_class/src/main.c:
```diff
@@ -968,46 +969,42 @@ static void test_traffic_class_recv_data_mix_all_2(void)
-}
+ZTEST(traffic_class_test_suite, test_traffic_class_tx_rx)
+{
+	test_traffic_class_setup_tx();
+	/* Send only same priority packets and verify that
+	 * all are sent with proper traffic class.
+	 */
+	test_traffic_class_send_data_prio_bk();
+	test_traffic_class_send_data_prio_be();
+	test_traffic_class_send_data_prio_ee();
+	test_traffic_class_send_data_prio_ca();
+	test_traffic_class_send_data_prio_vi();
+	test_traffic_class_send_data_prio_vo();
+	test_traffic_class_send_data_prio_ic();
+	test_traffic_class_send_data_prio_nc();
+	/* Then mix traffic classes and verify that higher
+	 * class packets are sent first.
+	 */
+	test_traffic_class_send_data_mix();
+	test_traffic_class_send_data_mix_all_1();
+	test_traffic_class_send_data_mix_all_2();
+	test_traffic_class_cleanup_tx();
+
+	test_traffic_class_setup_rx();
+	test_traffic_class_setup_recv();
+	test_traffic_class_recv_data_prio_bk();
+	test_traffic_class_recv_data_prio_be();
+	test_traffic_class_recv_data_prio_ee();
+	test_traffic_class_recv_data_prio_ca();
+	test_traffic_class_recv_data_prio_vi();
+	test_traffic_class_recv_data_prio_vo();
+	test_traffic_class_recv_data_prio_ic();
+	test_traffic_class_recv_data_prio_nc();
+	test_traffic_class_recv_data_mix();
+	test_traffic_class_recv_data_mix_all_1();
+	test_traffic_class_recv_data_mix_all_2();
+	test_traffic_class_cleanup_rx();
```
Are these all inter-dependent? Each test should be hermetic and run individually in any order.

hunterhu3000 replied:
These supposed to be independent, but actually not.
I have modified some code and made them independent. Thanks to review.

hunterhu3000 commented:
CONFIG_ZTEST_SHUFFLE can be set to y on this test case

hunterhu3000 committed:
tests: net: traffic_class move to new ztest API

Move traffic_class tests to use new ztest API
TEST=twister -T tests/net/traffic_class

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

enjiamai commented:
HI, @hunterhu3000 , sorry, I press the wrong button, it should be "rebase", could you please rebase and force-push it again, thanks!

hunterhu3000 commented:
> HI, @hunterhu3000 , sorry, I press the wrong button, it should be "rebase", could you please rebase and force-push it again, thanks!

Done
